ibis/backends/oracle/compiler.py

```diff
@@ -0,0 +1,604 @@
+        if how == "sample":
+            raise ValueError(
+                "Oracle only implements population correlation coefficient"
+            )
+        return self.agg.corr(left, right, where=where)
+
+    @visit_node.register(ops.Covariance)
+    def visit_Covariance(self, op, *, left, right, where, how):
+        if how == "sample":
+            return self.agg.covar_samp(left, right, where=where)
+        return self.agg.covar_pop(left, right, where=where)
+
+    @visit_node.register(ops.ApproxMedian)
+    def visit_ApproxMedian(self, op, *, arg, where):
+        return self.visit_Quantile(op, arg=arg, quantile=0.5, where=where)
+
+    @visit_node.register(ops.Quantile)
+    def visit_Quantile(self, op, *, arg, quantile, where):
+        suffix = "cont" if op.arg.dtype.is_numeric() else "disc"
+        funcname = f"percentile_{suffix}"
+
+        if where is not None:
+            arg = self.if_(where, arg)
+
+        expr = sge.WithinGroup(
+            this=self.f[funcname](quantile),
+            expression=sge.Order(expressions=[sge.Ordered(this=arg)]),
+        )
+        return expr
+
+    @visit_node.register(ops.CountDistinct)
+    def visit_CountDistinct(self, op, *, arg, where):
+        if where is not None:
+            arg = self.if_(where, arg)
+
+        return sge.Count(this=sge.Distinct(expressions=[arg]))
+
+    @visit_node.register(ops.CountStar)
+    def visit_CountStar(self, op, *, arg, where):
+        if where is not None:
+            return self.f.count(self.if_(where, 1, NULL))
+        return self.f.count(STAR)
+
+    @visit_node.register(ops.IdenticalTo)
+    def visit_IdenticalTo(self, op, *, left, right):
+        # sqlglot NullSafeEQ uses "is not distinct from" which isn't supported in oracle
+        return (
+            sg.case()
+            .when(left.eq(right).or_(left.is_(NULL).and_(right.is_(NULL))), 0)
+            .else_(1)
+            .eq(0)
+        )
+
+    @visit_node.register(ops.Xor)
+    def visit_Xor(self, op, *, left, right):
+        return (left.or_(right)).and_(sg.not_(left.and_(right)))
+
+    @visit_node.register(ops.TimestampTruncate)
+    @visit_node.register(ops.DateTruncate)
+    @visit_node.register(ops.TimeTruncate)
+    def visit_TimestampTruncate(self, op, *, arg, unit):
+        # FIXME: assertion errors
+        unit_mapping = {
+            "Y": "year",
+            "M": "month",
+            "W": "week",
+            "D": "day",
+            "h": "hour",
+            "m": "minute",
+            "s": "second",
+            "ms": "ms",
+            "us": "us",
+        }
+
+        if (unit := unit_mapping.get(unit.short)) in ("ms", "us", None):
+            raise com.UnsupportedOperationError(f"Unsupported truncate unit {unit}")
+
+        # The FuncGen for self.f.extract always resolves to sge.Extract
+        # and then complains about the missing expression argument
+        # so we construct it directly
+        return sge.Extract(this=unit, expression=arg)
```

Comment:
I see your comment, but I think perhaps you may have specified the order incorrectly. This should be doable as:

```python
self.f.extract(self.v[unit], arg)
```

`self.v[name]` constructs `sge.Var(this=name)` which is the sqlglot-ism for the `YEAR` syntax in `EXTRACT(YEAR FROM my_column)`.

Also, is `extract` the right operation here? This should be truncating a date/time/timestamp, not returning one of its components.

Reply:
it is not, should be `trunc`, at least for the datetimes